.github/workflows/drop-pr-caches.yml

```
@@ -0,0 +1,28 @@
+name: Drop PR Caches
+
+on:
+  pull_request_target:
+    types:
+      - closed
+
+jobs:
+  cleanup:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Cleanup
+        run: |
+          gh extension install actions/gh-actions-cache
+
+          cacheKeysForPR=$(gh actions-cache list -R $REPO -B $BRANCH -L 100 | cut -f 1)
+
+          # Turn off errexit to not fail the workflow while deleting cache keys.
+          set +e
+          echo "Deleting caches..."
+          for cacheKey in $cacheKeysForPR
+          do
+              gh actions-cache delete $cacheKey -R $REPO -B $BRANCH --confirm
+          done
```

```suggestion
          for i in {1..40}
          do
              cacheKeysForPR=$(gh actions-cache list -R $REPO -L 100 | cut -f 1 )
              for cacheKey in $cacheKeysForPR
              do
                  gh actions-cache delete "$cacheKey" -R $REPO --confirm
              done
          done
```
Deleting 100 caches won't even be close to enough. Even 4000 might not be enough in some cases, but it should handle even fairly heavy PRs.

---

Yes, but the default token only has 1000 api requests per hour. I think deleting the largest 100 entries will help significantly.